Issues with translation `json`s after #2271

## Description

<!-- Concisely describe the bug. Compare your experience with what you expected to happen. -->
<!-- For example: "I clicked the 'submit' button and instead of seeing a thank you message, I saw a blank page." -->
There are two problems with the translation `json`s after #2271 was merged:
1. The CI fails because some languages use the `nbsp` (non-breaking space) character, and it fails the linting step in the CI.
2. Some translations have still have the old kebab-cased keys that look like "###read-more###" in the GlotPress interface. We cannot ensure that the translators change the keys manually to camelCase, so we should add a step to camelize keys locally when saving jsons, instead.

## Reproduction

The first problem was reported by @zackkrida in #2398:

> While trying to push this I also noticed that after the merging of #2271, some of the generated translation files fail eslint:

```shell
openverse/frontend/src/locales/de.json
  171:29  error  Irregular whitespace not allowed  no-irregular-whitespace
  236:29  error  Irregular whitespace not allowed  no-irregular-whitespace
  472:29  error  Irregular whitespace not allowed  no-irregular-whitespace
  706:29  error  Irregular whitespace not allowed  no-irregular-whitespace

openverse/frontend/src/locales/fr.json
   68:28   error  Irregular whitespace not allowed  no-irregular-whitespace
   68:108  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:112  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:135  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:143  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:148  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:156  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:161  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:173  error  Irregular whitespace not allowed  no-irregular-whitespace
   71:32   error  Irregular whitespace not allowed  no-irregular-whitespace
   92:39   error  Irregular whitespace not allowed  no-irregular-whitespace
  182:48   error  Irregular whitespace not allowed  no-irregular-whitespace
  182:56   error  Irregular whitespace not allowed  no-irregular-whitespace

openverse/frontend/src/locales/lv.json
  11:24  error  Irregular whitespace not allowed  no-irregular-whitespace

openverse/frontend/src/locales/ru.json
  321:230  error  Irregular whitespace not allowed  no-irregular-whitespace
```

> I don't think we want to lint those files at all, so I've added them to the root .eslintignore. Does that seem right @obulat?

To see the second problem, you can run `just frontend/run i18n` and check one of the "<locale>.json" files. If you look at `nl.json`, and search for string "cc-search" or "read-more", you'll see that the key wasn't renamed to "ccSearch" or "readMore". This is why in the migration banner in `nl` locale (https://openverse.org/nl?referrer=creativecommons.org), you won't see "CC Search", or "read more" link.

## Additional context

<!-- Add any other context about the problem here; or delete the section entirely. -->
I've marked this issue as critical because the PRs are failing CI without it. 
<!-- If you would like to work on this, please comment below separately. -->

--- a/.eslintignore
+++ b/.eslintignore
@@ -7,3 +7,5 @@ frontend/storybook-static
 !frontend/.storybook
 
 !.pnpmfile.cjs
+
+frontend/src/locales/*.json

--- a/frontend/src/locales/scripts/utils.js
+++ b/frontend/src/locales/scripts/utils.js
@@ -37,6 +37,7 @@ const replacePlaceholders = (json) => {
   }
   if (typeof json === "string") {
     // Replace old kebab-cased keys that can still be in the translations with camelCased keys the app expects
+    // TODO: Remove this once all translation strings are updated.
     return json.replace(
       /###([a-zA-Z-]*)###/g,
       (_, match) => `{${camel(match)}}`

--- a/frontend/src/locales/scripts/utils.js
+++ b/frontend/src/locales/scripts/utils.js
@@ -1,6 +1,8 @@
 const { writeFile } = require("fs/promises")
 const os = require("os")
 
+const { camel } = require("case")
+
 /**
  * Mutates an object at the path with the value. If the path
  * does not exist, it is created by nesting objects along the
@@ -34,7 +36,11 @@ const replacePlaceholders = (json) => {
     return null
   }
   if (typeof json === "string") {
-    return json.replace(/###([a-zA-Z-]*)###/g, "{$1}")
+    // Replace old kebab-cased keys that can still be in the translations with camelCased keys the app expects
+    return json.replace(
+      /###([a-zA-Z-]*)###/g,
+      (_, match) => `{${camel(match)}}`
+    )
   }
   let currentJson = { ...json }
 

--- a/frontend/src/locales/scripts/utils.js
+++ b/frontend/src/locales/scripts/utils.js
@@ -25,6 +25,18 @@ exports.setToValue = function setValue(obj, path, value) {
   o[a[0]] = value
 }
 
+function replacer(_, match) {
+  // Replace ###<text>### from `po` files with {<text>} in `vue`.
+  // Additionally, the old kebab-cased keys that can still be in the
+  // translations are replaced with camelCased keys the app expects.
+  // TODO: Remove `camel` and warning once all translation strings are updated.
+  // https://github.com/WordPress/openverse/issues/2438
+  if (match.includes("-")) {
+    console.warn("Found kebab-cased key in translation strings:", match)
+  }
+  return `{${camel(match)}}`
+}
+
 /**
  * Replace ###<text>### with {<text>}.
  *
@@ -36,12 +48,7 @@ const replacePlaceholders = (json) => {
     return null
   }
   if (typeof json === "string") {
-    // Replace old kebab-cased keys that can still be in the translations with camelCased keys the app expects
-    // TODO: Remove this once all translation strings are updated.
-    return json.replace(
-      /###([a-zA-Z-]*)###/g,
-      (_, match) => `{${camel(match)}}`
-    )
+    return json.replace(/###([a-zA-Z-]*)###/g, replacer)
   }
   let currentJson = { ...json }
 

